tests/Mantis/ConfigParserTest.php

```diff
@@ -0,0 +1,432 @@
+
+# Mantis Core required for class autoloader and constants
+require_mantis_core();
+
+use PHPUnit_Framework_Constraint_IsType as PHPUnit_Type;
+
+
+/**
+ * Test cases for config API parser
+ *
+ * A list of test strings (as entered in adm_config_report page) will be parsed
+ * by MantisBT core, and the result compared to PHP's native interpretation
+ * using eval().
+ *
+ * @package    Tests
+ * @subpackage ConfigParser
+ */
+class Mantis_ConfigParserTest extends PHPUnit_Framework_TestCase {
+
+	/**
+	 * @var array List of test cases for scalar types
+	 */
+	private $cases_scalar = array();
+
+	/**
+	 * @var array List of test cases for arrays
+	 */
+	private $cases_array = array();
+
+	/**
+	 * Mantis_ConfigParserTest constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->initScalarTestCases();
+		$this->initArrayTestCases();
+	}
+
+	/**
+	 * Test a list of strings representing scalar values, making sure
+	 * the value and the type match
+	 *
+	 * @throws Exception
+	 */
+	public function testScalarTypes() {
+		foreach( $this->cases_scalar as $t_string => $t_expected_type ) {
+			$t_reference_result = eval( 'return ' . $t_string . ';' );
+
+			$t_parser = new ConfigParser( $t_string );
+			$t_parsed_result = $t_parser->parse();
+
+			$this->assertInternalType( $t_expected_type, $t_parsed_result );
+			$this->assertEquals( $t_parsed_result, $t_reference_result, $this->errorMessage( $t_string ) );
+		}
+	}
+
+	/**
+	 * Test various types of arrays
+	 *
+	 * @see initArrayTestCases
+	 * @throws Exception
+	 */
+	public function testArrays() {
+		foreach( $this->cases_array as $t_string ) {
+			$t_reference_result = eval( 'return ' . $t_string . ';' );
+
+			# Check that the parsed array matches the model array
+			$t_parser = new ConfigParser( $t_string );
+			$t_parsed_1 = $t_parser->parse();
+			$this->assertEquals( $t_parsed_1, $t_reference_result, $this->errorMessage( $t_string )  );
+
+			# Export converted array and parse again: result should match the model
+			$t_parser = new ConfigParser( var_export( $t_parsed_1 , true ) );
+			$t_parsed_2 = $t_parser->parse();
+			$this->assertEquals( $t_parsed_2, $t_reference_result, $this->errorMessage( $t_string )  );
+		}
+	}
+
+	/**
+	 * Test failure if we get extra tokens when the parser is set to error
+	 *
+	 * @throws Exception
+	 */
+	public function testExtraTokensError() {
+		$this->setExpectedExceptionRegExp('Exception', '/^Extra tokens found/');
+
+		$t_parser = new ConfigParser( '1; 2' );
+		$t_parser->parse( ConfigParser::EXTRA_TOKENS_ERROR );
+
+		$t_parser = new ConfigParser( 'array(); 2' );
+		$t_parser->parse( ConfigParser::EXTRA_TOKENS_ERROR );
+	}
+
+	/**
+	 * Test no errors if we get extra tokens when the parser is set to ignore
+	 *
+	 * @throws Exception
+	 */
+	public function testExtraTokensIgnore() {
+		$t_parser = new ConfigParser( '1; 2' );
+		$t_result = $t_parser->parse( ConfigParser::EXTRA_TOKENS_ERROR );
```

Comment:
You're right, this is a copy/paste error it should be *EXTRA_TOKENS_IGNORE*... And PHPUnit didn't catch it because there was an error in ConfigParser::parse(). I'll fix that.